### PR TITLE
[402] Due date on tasks

### PR DESCRIPTION
**Pull request by edwinthinks**
# 🔗 Issue
#402 

# ✍️ Description

Demo - https://www.loom.com/share/c7a2ec4e5216417e891d4d737d90959c

# 📷 Screenshots/Demos
Doing something fun this time and recording my process while implementing this - 

Part (1) - https://www.loom.com/share/0c7b71212c184367afdf9d44a0113d4a

### CODE REVIEW

**Comment by kasugaijin** [comment]
@edwinthinks standard failed fyi

**Comment by kasugaijin** [comment]
Looks good! I will watch the video when I get some time this weekend :)
Are you going to add the 'overdue' warning (https://github.com/rubyforgood/pet-rescue/issues/402) on the index and task/show? Or do you want that to be in a separate issue?

**Comment by edwinthinks** [comment]
Thanks @kasugaijin I think I'll include it in here in addition to include local_time to display the due dates

File: Gemfile [hunk 1]
```diff
@@ -83,6 +83,7 @@ gem "rails-i18n"
 # Provides helper methods to easily add 'active' tag on links
 gem "active_link_to"
 gem "nice_partials"
+gem "local_time"
```
Comment (edwinthinks) [inline Gemfile:86]: I didn't actually need this library but I'am thinking we may need this anyways eventually!

**Comment by edwinthinks** [comment]
@kasugaijin ready for re-review, I didn't include local_time like I mentioned before because i don't believe we actually need it yet.

**Comment by kasugaijin** [comment]
The walk through video is great @edwinthinks.
I think this would be good to link to in the ReadMe for new people on how to pick up an issue and make code changes. Thoughts?

**Comment by kasugaijin** [comment]
I agree that description on a task should not be required. I can see many cases where just a name alone will suffice. I have made an issue to remove this contraints on `Task` and `DefaultPetTask` and the `null: false` constraint on default_pet_tasks tabke.

**Comment by kasugaijin** [comment]
Hey @edwinthinks pipeline is not passing. I re-ran the pipeline a few times and it's the same system test. 

Also, when I checked out the branch to play with it the turbo stream is doing something weird with the inline for for me, on create and edit. Not sure why this would work on your machine and not mine.

pet tasks

**Comment by nsiwnf** [comment]
Reverted the local-time change and updated design to match figma

Screen Shot 2024-01-23 at 6 37 53 PM

**Review by nsiwnf — APPROVED** [review]
lgtm

**Pull request merged**